src/constructN.js

```diff
@@ -33,12 +31,24 @@ var nAry = require('./nAry');
  *      R.map(R.constructN(1, Widget), allConfigs); // a list of Widgets
  */
 module.exports = _curry2(function constructN(n, Fn) {
-    var f = function() {
-        var Temp = function() {}, inst, ret;
-        Temp.prototype = Fn.prototype;
-        inst = new Temp();
-        ret = Fn.apply(inst, arguments);
-        return Object(ret) === ret ? ret : inst;
-    };
-    return n > 1 ? curry(nAry(n, f)) : f;
+    if (n > 10) {
+        throw new Error('Constructor with greater than ten arguments');
+    }
+    if (n === 0) {
+        return function() { return new Fn(); };
+    }
+    return curry(nAry(n, function($0, $1, $2, $3, $4, $5, $6, $7, $8, $9) {
+        switch (arguments.length) {
+            case  1: return new Fn($0);
+            case  2: return new Fn($0, $1);
+            case  3: return new Fn($0, $1, $2);
+            case  4: return new Fn($0, $1, $2, $3);
+            case  5: return new Fn($0, $1, $2, $3, $4);
+            case  6: return new Fn($0, $1, $2, $3, $4, $5);
+            case  7: return new Fn($0, $1, $2, $3, $4, $5, $6);
+            case  8: return new Fn($0, $1, $2, $3, $4, $5, $6, $7);
+            case  9: return new Fn($0, $1, $2, $3, $4, $5, $6, $7, $8);
+            case 10: return new Fn($0, $1, $2, $3, $4, $5, $6, $7, $8, $9);
+        }
```

Comment:
Yes, but an earlier implementation, based on the Function constructor, did not have any particular length limitation.  It was good to be rid of it, even if it meant sacrificing a tiny bit of theoretic flexibility.  When we did that, we should have realized that the same sort of approach would also have fixed the issues with `construct`.  (But I might have the order of events muddled in my mind.)  I'm just glad you caught it now.